plugins/modules/monitoring/spectrum_maintenance.py

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2021 Uwe Waechter <uwe.waechter@scok.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: spectrum_maintenance
+short_description: put devices in/out maintenance or hibernating mode.
+description:
+   - Tested on CA Spectrum 10.4.3.x
+version_added: "1.0.0"
+author: "Uwe Waechter (@UweW)"
+options:
+    device:
+        aliases: [ host, name ]
+        required: true
+        description:
+            - IP address of the device.
+            - If a hostname is given, it will be resolved to the IP address.
+        type: str
+    state:
+        description:
+            - On C(inMaintenence) put device in maintenence mode
+            - On C(inHibernation) put device in hibernation mode
+            - On C(isManaged) put device in isManaged mode
+        choices: ['inMaintenance', 'inHibernation', 'isManaged']
+        default: 'isManaged'
+        type: str
+    url:
+        aliases: [ oneclick_url ]
+        required: true
+        description:
+            - HTTP, HTTPS URL of the Oneclick server in the form (http|https)://host.domain[:port]
+        type: str
+    url_username:
+        aliases: [ oneclick_user ]
+        required: true
+        description:
+            - Oneclick user name.
+        type: str
+    url_password:
+        aliases: [ oneclick_password ]
+        required: true
+        description:
+            - Oneclick user password.
+        type: str
+    use_proxy:
+        required: false
+        description:
+            - if C(no), it will not use a proxy, even if one is defined in an environment
+                variable on the target hosts.
+        default: 'no'
+        type: bool
+    validate_certs:
+        required: false
+        description:
+            - If C(no), SSL certificates will not be validated. This should only be used
+                on personally controlled sites using self-signed certificates.
+        default: 'no'
+        type: bool
+notes:
+   -  expects IP address as inventory_hostname
+'''
+
+EXAMPLES = '''
+- name: put device in Maintenance mode
```

Comment:
```suggestion
- name: Put device in Maintenance mode
```
Task names should start with an uppercase letter.